File: src/univ_prop/cpp/coef.cpp
#include <iostream>
#include <cmath>

double f(double chi, double r0, double C){
    return 1 - (chi * chi) / (r0) - C;
}

double d_f(double chi, double r0, double r, double S, double z, double mu) {
    return ((std::sqrt(mu)) / (r * r0)) * chi * (z * S - 1);
}

double g(double chi, double mu, double S, double dt){
    return dt - (1 / (std::sqrt(mu))) * chi * chi * chi * S;
}

double d_g(double chi, double r, double C){
    return 1 - ((chi * chi) / r) * C;
}

File: src/univ_prop/pyuniversal/coef.py
import numpy as np
from numpy.linalg import norm
from .stumpff import *

def f(chi,r0,C):
    return 1 - (chi**2 / r0) * C

def g(dt,chi,mu,S):
    return dt - (1 / np.sqrt(mu)) * chi**3 * S

def df(chi,r0,r,S,z,mu):
    return (np.sqrt(mu) / (r * r0)) * chi * (z * S - 1)

def dg(chi,r,C):
    return 1 - (chi**2 / r) * C
